src/State/Processor/SerializeProcessor.php

```diff
@@ -57,10 +57,12 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
             trigger_deprecation('api-platform/core', '3.3', 'The resource class on the router is not the same as the operation\'s class which leads to wrong behaviors. Prefer using "stateOptions" if you need to change the entity class.');
         }
 
-        $serializerContext = $this->serializerContextBuilder->createFromRequest($request, true, [
-            'resource_class' => $class,
-            'operation' => $operation,
-        ]);
+        if (null === ($serializerContext = $request->attributes->get('_api_normalization_context'))) {
```

Comment:
shouldn't we move the responsibility of this to the `SerializerContextBuilder` directly? It'd avoid duplicating this condition. I'd also do the `set` there.

Reply:
Do you have in mind a decorator of `api_platform.serializer.context_builder` with a low priority to `get`/`set` the computed context in the attribute request ?
It was certainly a better way to go because the `SerializerContextBuilder` is at the end of the inheritance chain of `api_platform.serializer.context_builder`.

Reply:
Indeed, we could introduce a RequestCacheSerializerContextBuilder that has a negative priority?

Reply:
I'll take care of it.
What about denormalizationContext? This notion doesn't exist in the request's attributes and I see that we can have 2 calls as well:
- https://github.com/api-platform/core/blob/fbb53e5e35ca0ec3de26ddc7de7ea4d1dda5c20b/src/Symfony/Bundle/ArgumentResolver/PayloadArgumentResolver.php#L78
- and in deserialize layer : https://github.com/api-platform/core/blob/fbb53e5e35ca0ec3de26ddc7de7ea4d1dda5c20b/src/Symfony/EventListener/DeserializeListener.php#L136 or https://github.com/api-platform/core/blob/fbb53e5e35ca0ec3de26ddc7de7ea4d1dda5c20b/src/State/Provider/DeserializeProvider.php#L68

I'm missing information about the execution level of the PayloadArgumentResolver, can you tell me more? Are we adding this support to the `RequestCacheSerializerContextBuilder` as well ?